src/main/java/gg/uhc/uhc/inventory/IconInventory.java

```diff
@@ -68,14 +96,7 @@ protected int add(IconStack icon) {
         // add item to list in the correct location
         final int index = indexToInsert(icon);
         icons.add(index, icon);
-
-        // make sure this is enough space now
-        ensureInventorySize();
-
-        // rerender all items past this one (inclusive) as they have shifted index right 1
-        for (int i = index; i < icons.size(); i++) {
-            inventory.setItem(i, icons.get(i));
-        }
+        reopenForCurrentViewers();
```

Comment:
Instead of reopening in here and weight update can we loop the inventories and just set the items required with the old logic instead?

```java
for (Inventory inv : openedInventories) {
    for (int i = index; i < icons.size(); i++) {
        inventory.setItem(i, icons.get(i));
    }
}
```